TodoApi/Extensions/SerilogExtensions.cs

```diff
@@ -0,0 +1,68 @@
+using Microsoft.Extensions.Options;
+using Serilog;
+using Serilog.Exceptions;
+
+namespace TodoApi;
+
+public static class SerilogExtensions
+{
+    public static WebApplicationBuilder AddSerilog(
+        this WebApplicationBuilder builder,
+        string sectionName = "Serilog")
+    {
+        builder.Services.AddOptions<SerilogOptions>()
+            .BindConfiguration(sectionName)
+            .ValidateDataAnnotations();
```

Comment:
Since this is only used in this the UseSerilog call, it might be cleaner to Bind the configuration here.

```suggestion
        var serilogOptions = new SerilogOptions();
        builder.Configuration.GetSection(sectionName).Bind(serilogOptions);
```

Then we can remove some noise from the `UseSerilog` stuff below. Which btw isn't wrong, it just seems like there's no value in using options here.

Reply:
I should mention thought, the benefit of using `IOptions<T>` is letting other code configure this as well (via calls to AddOptions\<T\> or Configure\<T\>). I think that might be overkill here.